spring-security-modules/spring-security-web-boot-4/src/main/java/com/baeldung/apikeysecretauth/SpringWebSecurityConfig.java

```diff
@@ -0,0 +1,59 @@
+package com.baeldung.apikeysecretauth;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import com.baeldung.apikeysecretauth.repository.UserKeysRepository;
+import com.baeldung.apikeysecretauth.security.ApiKeySecretAuthenticationProvider;
+import com.baeldung.apikeysecretauth.security.ApiKeySecretFilter;
+import com.baeldung.apikeysecretauth.security.RestAuthenticationEntryPoint;
+
+@EnableWebSecurity
+public class SpringWebSecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserKeysRepository userKeysRepository;
+
+    public SpringWebSecurityConfig(UserKeysRepository userKeysRepository) {
+        this.userKeysRepository = userKeysRepository;
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        final AuthenticationEntryPoint restAuthenticationEntryPoint = new RestAuthenticationEntryPoint();
+
+        http.sessionManagement()
+            .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+            .and()
+            .authorizeRequests()
+            .anyRequest()
+            .authenticated()
+            .and()
+            .addFilterBefore(new ApiKeySecretFilter(authenticationManager(), restAuthenticationEntryPoint), BasicAuthenticationFilter.class)
+            .exceptionHandling()
+            .authenticationEntryPoint(restAuthenticationEntryPoint);
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) {
+        auth.authenticationProvider(new ApiKeySecretAuthenticationProvider(passwordEncoder(), userKeysRepository));
+    }
+
+    @Bean
```

Comment:
not needed here. but, will be needed in updated code since test cases mock the PasswordEncoder and having it as a spring managed bean makes it easier to mock it.
let me know if you prefer an alternative approach